### PR TITLE
Add sound null-safety

### PR DESCRIPTION
**Pull request by figengungor**

### CODE REVIEW

**Comment by pr-1** [comment]
@sarbagyastha Any updates on this ?

File: packages/youtube_player_flutter/pubspec.yaml [hunk 1]
```diff
@@ -5,18 +5,18 @@ repository: https://github.com/sarbagyastha/youtube_player_flutter
 homepage: https://github.com/sarbagyastha/youtube_player_flutter/tree/master/packages/youtube_player_flutter
 
 environment:
-  sdk: '>=2.7.0 <3.0.0'
+  sdk: '>=2.12.0 <3.0.0'
   flutter: '>=1.12.13+hotfix.5'
```
Comment (pr-1) [inline packages/youtube_player_flutter/pubspec.yaml:9]: Update this to 
`flutter: '>=2.0.1'`

File: packages/youtube_player_iframe/pubspec.yaml [hunk 2]
```diff
@@ -5,15 +5,15 @@ repository: https://github.com/sarbagyastha/youtube_player_flutter
 homepage: https://github.com/sarbagyastha/youtube_player_flutter/tree/master/packages/youtube_player_iframe
 
 environment:
-  sdk: '>=2.7.0 <3.0.0'
+  sdk: '>=2.12.0 <3.0.0'
   flutter: '>=1.22.0 <2.0.0'
```
Comment (pr-1) [inline packages/youtube_player_iframe/pubspec.yaml:9]: Similarly here 
`flutter: '>2.0.1'`

**Review by pr-1 — CHANGES_REQUESTED** [review]
Update in pubspec : 
packages/youtube_player_flutter/pubspec.yaml
`flutter : '>2.0.1'`

File: packages/youtube_player_flutter/pubspec.yaml (hunk 1 above)
Comment (figengungor) [reply packages/youtube_player_flutter/pubspec.yaml]: Done.

File: packages/youtube_player_iframe/pubspec.yaml (hunk 2 above)
Comment (figengungor) [reply packages/youtube_player_iframe/pubspec.yaml]: Done.

**Comment by pr-1** [comment]
update version of flutter_inappwebview to latest in pubspec.

**Comment by pr-1** [comment]
@figengungor Let me know if you're on it else I will start.

**Comment by pives** [comment]
i'd love to avoid the gradle update you put in the PR as appcenter isn't compatible with latest. @figengungor

**Comment by figengungor** [comment]
I fixed dart sdk constraints, updated dependencies which are mistakenly reverted back and downgraded gradle versions.

**Pull request merged**